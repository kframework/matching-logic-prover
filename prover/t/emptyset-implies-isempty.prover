claim \implies( \and(\equals(variable("S") { SetInt }, emptyset), .Patterns)
              , \and(isEmpty(variable("S") { SetInt }, .Patterns), .Patterns)
              )
strategy search-bound(2) 
// strategy right-unfold ; simplify ; instantiate-existentials ; smt-z3
