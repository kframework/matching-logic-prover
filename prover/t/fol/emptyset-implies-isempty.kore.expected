<prover>
  <k>
    .
  </k>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      active: true, id: root, parent: .
      <k>
        \implies ( \and ( \equals ( S { SetInt } , emptyset ) , .Patterns ) , \exists { .Patterns } \and ( isEmpty ( S { SetInt } , .Patterns ) , .Patterns ) )
      </k>
      <strategy>
        success
      </strategy>
      <expected>
        success
      </expected>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom \forall { S { SetInt } , .Patterns } \iff-lfp ( isEmpty ( S { SetInt } , .Patterns ) , \or ( \exists { .Patterns } \and ( \equals ( S { SetInt } , emptyset ) , .Patterns ) , .Patterns ) )
    </declaration> <declaration>
      axiom functional ( isEmpty )
    </declaration> <declaration>
      symbol isEmpty ( SetInt , .Sorts ) : Bool
    </declaration>
  </declarations>
</prover>
